api/src/state.rs

```diff
@@ -385,4 +427,53 @@ impl StateApi {
             }
         }
     }
+
+    /// Retrieve table item for a specific ledger version
+    pub fn raw_table_item(
+        &self,
+        accept_type: &AcceptType,
+        table_handle: Address,
+        table_item_request: RawTableItemRequest,
+        ledger_version: Option<U64>,
+    ) -> BasicResultWith404<MoveValue> {
+        // Retrieve local state
+        let (ledger_info, ledger_version, state_view) =
+            self.preprocess_request(ledger_version.map(|inner| inner.0))?;
+
+        let state_key = StateKey::table_item(
+            TableHandle(table_handle.into()),
+            table_item_request.key.0.clone(),
+        );
+        let bytes = state_view
+            .get_state_value(&state_key)
+            .context(format!(
+                "Failed when trying to retrieve table item from the DB with key: {}",
+                table_item_request.key,
+            ))
+            .map_err(|err| {
+                BasicErrorWith404::internal_with_code(
+                    err,
+                    AptosErrorCode::InternalError,
```

Comment:
I don't think this is internal error? the db returns error if it's pruned

Reply:
I just copied this logic from  [https://github.com/aptos-labs/aptos-core/blob/c4137efcba944be17247e4b21e470180f1b80777/api/src/state.rs#L415](line 415). Not sure why the error should be different from previous one?

Reply:
hmm, that's also a problem